applications/OptimizationApplication/python_scripts/responses/combined_response_function.py

```diff
@@ -0,0 +1,139 @@
+from typing import Optional
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.OptimizationApplication as KratosOA
+from KratosMultiphysics.OptimizationApplication.responses.response_function import ResponseFunction
+from KratosMultiphysics.OptimizationApplication.utilities.union_utilities import SupportedSensitivityFieldVariableTypes
+from KratosMultiphysics.OptimizationApplication.utilities.model_part_utilities import ModelPartOperation
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_problem import OptimizationProblem
+from KratosMultiphysics.OptimizationApplication.utilities.component_data_view import ComponentDataView
+
+def Factory(model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem) -> ResponseFunction:
+    if not parameters.Has("name"):
+        raise RuntimeError(f"CombinedResponseFunction instantiation requires a \"name\" in parameters [ parameters = {parameters}].")
+    if not parameters.Has("settings"):
+        raise RuntimeError(f"CombinedResponseFunction instantiation requires a \"settings\" in parameters [ parameters = {parameters}].")
+
+    return CombinedResponseFunction(parameters["name"].GetString(), model, parameters["settings"], optimization_problem)
+
+class CombinedResponseFunction(ResponseFunction):
+    """Response function to combine multiple response functions.
+
+    This response function can combine multiple response functions to a single response function.
+    Following methods are used to combine the response values:
+              1. weighted sum - Weighted summation of responses list is used as the combined response function value.
+
+    """
+    def __init__(self, name: str, model: Kratos.Model, parameters: Kratos.Parameters, optimization_problem: OptimizationProblem):
+        super().__init__(name)
+
+        default_settings = Kratos.Parameters("""{
+            "combining_method"   : "sum",
+            "combining_responses": [
+                {
+                    "response_name"  : "",
+                    "response_weight": 0.0
+                }
+            ]
+        }""")
+        parameters.ValidateAndAssignDefaults(default_settings)
+        self.combining_method = parameters["combining_method"].GetString()
+
+        self.model = model
+        self.model_part: Optional[Kratos.ModelPart] = None
+        self.optimization_problem = optimization_problem
+
+        self.list_of_responses: 'list[tuple[ResponseFunction, float]]' = []
+        for response_params in parameters["combining_responses"].values():
+            response_params.ValidateAndAssignDefaults(default_settings["combining_responses"].values()[0])
+            self.list_of_responses.append(
+                (
+                    self.optimization_problem.GetResponse(response_params["response_name"].GetString()),
```

Comment:
How do you make sure that this response is called after other ones are already added to the optimization_problem?

Reply:
We cannot. Responses should be added in the order which they are used. It is the users responsibility.

Reply:
Then, add a descriptive error so that the user can understand why it doesn't work (if the response is not found in the optimization_problem object)

Reply:
done :)